### PR TITLE
Added `edge_index` property to `TemporalData`

### PR DESCRIPTION
**Pull request by SalvishGoomanee**
Added the `edge_index` property in order to make it easier to visualize time series of temporal by simply doing a `test_data.edge_index` and using `networkx` which is useful for large multiple graph datasets.

### CODE REVIEW

File: torch_geometric/data/temporal.py [hunk 1]
```diff
@@ -212,6 +212,10 @@ def num_edges(self) -> int:
         r"""Alias for :meth:`~torch_geometric.data.TemporalData.num_events`."""
         return self.num_events
 
+    @property
+    def edge_index(self) -> Any:
```
Comment (EdisonLeeeee) [inline torch_geometric/data/temporal.py:216]:
```suggestion
    def edge_index(self) -> Optional[Tensor]:
```

File: torch_geometric/data/temporal.py [hunk 2]
```diff
@@ -212,6 +212,10 @@ def num_edges(self) -> int:
...
+        return self['edge_index'] if 'edge_index' in self._store else None
```
Comment (EdisonLeeeee) [inline torch_geometric/data/temporal.py:217]: `edge_index` can be also obtained by stacking `src` and `dst`, WDYT?

Comment (SalvishGoomanee) [reply torch_geometric/data/temporal.py:217]: This is in order to build a graph based on temporal edges by creating a TemporalData that networkx can understand just like with the Data object. I agree, one can stack `src` and `dst' but this is just a bit cleaner.

Comment (EdisonLeeeee) [reply torch_geometric/data/temporal.py:217]: Yep, I mean we can do something like:
```python
    @property
    def edge_index(self):
        edge_index = self['edge_index'] 
        if edge_index is None and not (self.src is None or self.dst is None):
            edge_index = torch.stack([self.src, self.dst], dim=0)
        return edge_index
```

**Review by rusty1s — APPROVED** [review]

**Pull request merged**